app/client/cli/actor.go

```diff
@@ -0,0 +1,217 @@
+		cmds = append(cmds, cmd)
+	}
+
+	return cmds
+}
+
+func newActorCommands(cmdDef actorCmdDef) (cmds []*cobra.Command) {
+	stakeCmd := &cobra.Command{
+		Use:   "Stake <fromAddr> <amount> <RelayChainIDs> <serviceURI>",
+		Short: "Stake a node in the network. Custodial stake uses the same address as operator/output for rewards/return of staked funds.",
+		Long: `Stake the node into the network, making it available for service.
+Will prompt the user for the <fromAddr> account passphrase. If the node is already staked, this transaction acts as an *update* transaction.
+A node can updated relayChainIDs, serviceURI, and raise the stake amount with this transaction.
+If the node is currently staked at X and you submit an update with new stake Y. Only Y-X will be subtracted from an account
+If no changes are desired for the parameter, just enter the current param value just as before`,
+		Args: cobra.ExactArgs(4),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// TODO(pocket/issues/150): update when we have keybase
+			pk, err := readEd25519PrivateKeyFromFile(privateKeyFilePath)
+			if err != nil {
+				return err
+			}
+			// currently ignored since we are using the address from the PrivateKey
+			// fromAddr := crypto.AddressFromString(args[0])
+			amount := args[1]
+
+			am, err := sharedTypes.StringToBigInt(args[1])
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			if sharedTypes.BigIntLessThan(am, stakingRecommendationAmount) {
+				fmt.Printf("The amount you are staking for is below the recommendation of %d POKT, would you still like to continue? y|n\n", stakingRecommendationAmount.Div(stakingRecommendationAmount, big.NewInt(1000000)).Int64())
+				if !Confirmation(pwd) {
+					return fmt.Errorf("aborted")
+				}
+			}
+
+			reg, err := regexp.Compile("[^,a-fA-F0-9]+")
+			if err != nil {
+				log.Fatal(err)
+			}
+			rawChains := reg.ReplaceAllString(args[2], "")
+			chains := strings.Split(rawChains, ",")
+			serviceURI := args[3]
+
+			fmt.Println("Enter Passphrase: ")
+			// TODO (team): passphrase is currently not used since there's no keybase yet, the prompt is here to mimick the real world UX
+			_ = Credentials(pwd)
+
+			msg := &types.MessageStake{
+				PublicKey:     pk.PublicKey().Bytes(),
+				Chains:        chains,
+				Amount:        amount,
+				ServiceUrl:    serviceURI,
+				OutputAddress: []byte{},     // TODO(deblasis): 👀
+				Signer:        pk.Address(), // TODO(deblasis): figure out where I should get this from (https://github.com/pokt-network/pocket/pull/169#discussion_r953186602)
+				ActorType:     cmdDef.ActorType,
+			}
+
+			codec := sharedTypes.GetCodec()
+			anyMsg, err := codec.ToAny(msg)
+
+			signature, _ := pk.Sign(msg.GetSignBytes())
+
+			tx := &utilityTypes.Transaction{
+				Msg: anyMsg,
+				Signature: &utilityTypes.Signature{
+					Signature: signature,
+					PublicKey: pk.PublicKey().Bytes(),
+				},
+				Nonce: "", // TODO(deblasis): figure out where I should get this from
```

Comment:
As currently specified, the nonce is just a random number

Reply:
Addressed in https://github.com/pokt-network/pocket/pull/177/commits/5e185e47f1fa61fe479a687d9058492d8870c794